xpra/codecs/nvenc/encoder.pyx

```diff
@@ -2357,8 +2646,16 @@ cdef class Encoder:
                 log("GPU memcopy %i bytes from %#x to %#x", stride*h, int(gpu_buffer), int(self.cudaInputBuffer))
             else:
                 stride = self.copy_image(image, False)
+                log("memcpy_htod(cudaInputBuffer=%s, inputBuffer=%s)", self.cudaInputBuffer, self.inputBuffer)
                 driver.memcpy_htod(self.cudaInputBuffer, self.inputBuffer)
-            self.exec_kernel(cuda_context, w, h, stride)
+            try:
+                self.exec_kernel(cuda_context, w, h, stride)
+            except LogicError:
+                if USE_SINGLETON_ENCODER:
+                    log("got kernel error exception, cleaning global instance")
+                    Encoder.clean_instance()
```

Comment:
Is this going to re-initialize nvenc or just the encoder context?
Do we want to do this for the non singleton case too?

Reply:
this doesn't reload the nvenc libraries in this case - just cleaning up the encoder context through the usual `clean()`, `cuda_clean()`, etc. and letting it reinitialize on the next frame/attempt. While testing this out, I kept running into issues where the kernel wouldn't init correctly and break any future attempts to encode. It was easier to just dump the whole thing and let it start over with a new encoder context.
It probably would be good to detect if this is happening too frequently and do something different, but so far not seeing it happen too much.